src/editorial/web/components/lines/styles.ts

```diff
@@ -2,53 +2,60 @@ import { css } from '@emotion/react';
 import { squigglyImage, height as squigglyImageHeight } from './squiggly';
 import { dottedImage, height as dottedImageHeight } from './dotted';
 import { dashedImage, height as labsImageHeight } from './dashed';
-import { line } from '@guardian/src-foundations/palette';
 import { remSpace } from '@guardian/src-foundations';
 import { LineCount } from '.';
 
 const lineGap = remSpace[1];
 
-export const straightLines = css`
-	background-image: repeating-linear-gradient(
-		to bottom,
-		${line.primary},
-		${line.primary} 1px,
-		transparent 1px,
-		transparent ${lineGap}
-	);
-	background-repeat: repeat-x;
-	background-position: top;
-`;
-
-export const oneLine = css`
-	background-size: 1px;
-	height: 1px;
-`;
-
-export const fourLines = css`
-	background-size: 1px calc(${lineGap} * 3 + 1px);
-	height: calc(${lineGap} * 3 + 1px);
-`;
-
-export const eightLines = css`
-	background-size: 1px calc(${lineGap} * 7 + 1px);
-	height: calc(${lineGap} * 7 + 1px);
-`;
-
-export const squigglyLines = (count: LineCount) => css`
-	background-image: url(${squigglyImage(count)});
+export const straightLines = (count: LineCount, color: string) => {
+	const baseStyles = css`
+		background-image: repeating-linear-gradient(
+			to bottom,
+			${color},
+			${color} 1px,
+			transparent 1px,
+			transparent ${lineGap}
+		);
+		background-repeat: repeat-x;
+		background-position: top;
+	`;
+
+	switch (count) {
+		case 1:
+			return css`
+				${baseStyles};
+				background-size: 1px;
+				height: 1px;
+			`;
+		case 4:
+			return css`
+				${baseStyles};
+				background-size: 1px calc(${lineGap} * 3 + 1px);
+				height: calc(${lineGap} * 3 + 1px);
+			`;
+		case 8:
+			return css`
+				${baseStyles};
+				background-size: 1px calc(${lineGap} * 7 + 1px);
+				height: calc(${lineGap} * 7 + 1px);
+			`;
```

Comment:
This could be simplified to handle any number other than 1.

```ts
		default:
			return css`
				${baseStyles};
				background-size: 1px calc(${lineGap} * ${count-1} + 1px);
				height: calc(${lineGap} * ${count-1} + 1px);
			`;
```

Reply:
Oh nice! I would love to see that improvement to the api.

Previously, we only thought of Lines as coming in batches of 4 or 8 but in truth it would be a lot easier to have a more open api - we actually use single 'lines' in lots of places (the article meta is a good example). We get around this by using borders but it'd be great to move away from that.

Reply:
https://github.com/guardian/source/pull/790 ?